onnxruntime/core/mlas/lib/quantize.cpp

```diff
@@ -263,6 +469,60 @@ Return Value:
     }
 }
 
+
+template<typename DataType>
+void
+MLASCALL
+MlasQLinearAddKernel(
+    const DataType* InputA,
+    float ScaleA,
+    DataType ZeroPointA,
+    const DataType* InputB,
+    float ScaleB,
+    DataType ZeroPointB,
+    float ScaleC,
+    DataType ZeroPointC,
+    DataType* OutputC,
+    size_t LengthA,
+    size_t LengthB
+    )
+{
+    constexpr int32_t MinimumValue = std::numeric_limits<DataType>::min();
+    constexpr int32_t MaximumValue = std::numeric_limits<DataType>::max();
+
+    size_t N = std::max(LengthA, LengthB);
+    if (LengthA == 1) {
+        float ValueA = ScaleA * (int(InputA[0]) - int(ZeroPointA));
+        for (size_t n = 0; n < N; n++) {
+            float ValueB = ScaleB * (int(InputB[n]) - int(ZeroPointB));
+            int32_t IntValueC = (int32_t)std::nearbyintf((ValueA + ValueB) / ScaleC) + int32_t(ZeroPointC);
+            IntValueC = std::max(IntValueC, MinimumValue);
```

Comment:
looks like the different branches have a lot of code duplication: make a helper template for that?

Reply:
Correct. Same as above.

Reply:
Nice! Fixed!